File: TransCoder_tokenized_test_set_functions/java/NUMBER_OF_SUBSTRINGS_WITH_ODD_DECIMAL_VALUE_IN_A_BINARY_STRING.java
static int countSubstr ( String s ) {
  int n = s . length ( );
  int [ ] auxArr = new int [ n ];
  if ( s . charAt ( 0 ) == '1' ) auxArr [ 0 ] = 1;
  for ( int i = 1;
  i < n;
  i ++ ) {
    if ( s . charAt ( i ) == '1' ) auxArr [ i ] = auxArr [ i - 1 ] + 1;
    else auxArr [ i ] = auxArr [ i - 1 ];
  }
  int count = 0;
  for ( int i = n - 1;
  i >= 0;
  i -- ) if ( s . charAt ( i ) == '1' ) count += auxArr [ i ];
  return count;
}

File: results/java/NUMBER_OF_SUBSTRINGS_WITH_ODD_DECIMAL_VALUE_IN_A_BINARY_STRING.cpp
int f_filled ( string s ) {
  int n = s . length ( ) ;
  vector < int > auxArr ;
  auxArr . reserve ( n ) ;
  if ( s [ 0 ] == '1' ) {
    auxArr . push_back ( 1 ) ;
  }
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    if ( s [ i ] == '1' ) {
      auxArr [ i ] = auxArr [ i - 1 ] + 1 ;
    }
    else {
      auxArr [ i ] = auxArr [ i - 1 ] ;
    }
  }
  int count = 0 ;
  for ( int i = n - 1 ;
  i >= 0 ;
  i -- ) if ( s [ i ] == '1' ) count += auxArr [ i ] ;
  return count ;
}
